.circleci/config.yml

```diff
@@ -15,28 +15,55 @@ jobs:
     steps:
       - checkout
       - restore_cache:
+          name: Restore cache (backend)
           keys:
           - v1-dependencies-{{ checksum "src/server/package.json" }}
           # fallback to using the latest cache if no exact match is found
           - v1-dependencies-
 
+      - restore_cache:
+          name: Restore cache (frontend)
+          keys:
+          - v1-dependencies-{{ checksum "src/client/package.json" }}
+          # fallback to using the latest cache if no exact match is found
+          - v1-dependencies-
+
       - run: 
-          name: Deployment
+          name: Deployment (backend)
           command: cd src/server && npm install
 
+      - run: 
+          name: Deployment (frontend)
+          command: cd src/client && npm install
+
       - save_cache:
+          name: Saving cache (backend)
           paths:
             - src/server/node_modules
           key: v1-dependencies-{{ checksum "src/server/package.json" }}
 
+      - save_cache:
+          name: Saving cache (frontend)
+          paths:
+            - src/client/node_modules
+          key: v1-dependencies-{{ checksum "src/client/package.json" }}
+
       - run:
-          name: Static Analysis (EsLint)
-          command: node ./src/server/node_modules/eslint/bin/eslint.js -c ./src/server/.eslintrc.json --ignore-path ./src/server/.eslintignore ./src/server
+          name: Static Analysis (backend)
+          command: cd src/server && npm run eslint
 
       - run:
-          name: Mocha tests
+          name: Static Analysis (frontend)
+          command: cd src/client && npm run eslint
+
+      - run:
+          name: Mocha tests (backend)
           command: cd src/server/ && npm run test-coverage
 
+      - run:
+          name: Jest tests (frontend)
+          command: cd src/client/ && npm run test-coverage
```

Comment:
https://circleci.com/gh/michaelKurowski/lokim2/288?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link
![image](https://user-images.githubusercontent.com/2772942/39956610-6dd3705c-55e4-11e8-9abe-182ed2086b52.png)
Why are they throwing?

Reply:
They're throwing as eslint wouldn't pass without type protection of props in react. I purposefully test the components in my tests to load without props without crashing so that they prove to be robust. Thus, these errors are thrown, as the types are "incorrect". So it's either we keep the errors, I remove the tests, or we ignore prop types from eslint.

Reply:
It's Jest errors, not eslint.

Reply:
Yup, because the props are not receiving the expected type, as per PropTypes.

Reply:
Removed.